src/webview/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -435,6 +450,26 @@ public void onReceivedError(WebView view, WebResourceRequest request, WebResourc
 							"}", error.getErrorCode(), error.getDescription()), false);
 				}
 			}
+
+			@Override
+			public void onPageFinished(WebView view, String url) {
+				trace(this, "onPageFinished() :: url: %s, isMigrationRunning: %s", url, isMigrationRunning);
+				if (isMigrationRunning && url.contains("/login")) {
+					isMigrationRunning = false;
```

Comment:
If the url does not contain `/login` that means the cookie has been found, right? If so, I think it makes sense to set `isMigrationRunning = false` so we never have to check the URL again, eg:

```
				if (isMigrationRunning) {
					isMigrationRunning = false;
					 if (url.contains("/login")) {
```

Reply:
This method is called with the flag `isMigrationRunning = true` with other URLs than `/login` before, so changing that would cause to set the variable `false` before reach the point desired.